src/rdf.rs

```diff
@@ -466,6 +519,190 @@ impl Graph {
     pub fn add(&mut self, triple: Triple) {
         self.triples.push(triple);
     }
+
+    pub fn as_ref(&self) -> GraphRef {
+        GraphRef {
+            triples: self.triples.iter().collect(),
+        }
+    }
+}
+
+impl<'a> GraphRef<'a> {
+    /// Remove triple matching given terms
+    pub fn take(
```

Comment:
I would rename it to `take_out` or something that explicitly say it will get removed

Reply:
Isn't `take` used to mean removing a value?
https://doc.rust-lang.org/std/option/enum.Option.html#method.take
https://docs.serde.rs/serde_json/enum.Value.html#method.take

Admittedly this is different because it involves filtering, rather than taking out the whole value - and it uses references rather than owned values. It's maybe more like a `drain_filter`, except not using a function, and only takes the first value. (Originally I had it return a Vec and then another function to take one, but then didn't need the vector one.)
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter
https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.drain_filter
Maybe `take_select_first`...?

Reply:
Ah fair enough, `take` sounds good, consistency is key :smile: